src/rabbit_shovel_dyn_worker_sup.erl

```diff
@@ -31,12 +31,26 @@ start_link(Name, Config) ->
 %%----------------------------------------------------------------------------
 
 init([Name, Config0]) ->
-    Config = rabbit_data_coercion:to_proplist(Config0),
+    Config  = rabbit_data_coercion:to_proplist(Config0),
+    Delay   = pget(<<"reconnect-delay">>, Config, ?DEFAULT_RECONNECT_DELAY),
+    case Name of
+      {VHost, ShovelName} -> rabbit_log:debug("Shovel '~s' in virtual host '~s' will use reconnection delay of ~p", [ShovelName, VHost, Delay]);
+      ShovelName          -> rabbit_log:debug("Shovel '~s' will use reconnection delay of ~s", [ShovelName, Delay])
+    end,
+    Restart = case Delay of
+        N when is_integer(N) andalso N > 0 ->
+          case pget(<<"src-delete-after">>, Config, <<"never">>) of
```

Comment:
Needs to handle `queue-length` value - `After initial lenght transferred` in UI

Reply:
We now handle it in the Shovel API so let's do that in a separate PR.